binder/binder.go

```diff
@@ -10,15 +11,69 @@ var (
 	ErrMapNotConvertable       = errors.New("binder: map is not convertable to map[string]string or map[string][]string")
 )
 
-// Init default binders for Fiber
-var (
-	HeaderBinder     = &headerBinding{}
-	RespHeaderBinder = &respHeaderBinding{}
-	CookieBinder     = &cookieBinding{}
-	QueryBinder      = &queryBinding{}
-	FormBinder       = &formBinding{}
-	URIBinder        = &uriBinding{}
-	XMLBinder        = &xmlBinding{}
-	JSONBinder       = &jsonBinding{}
-	CBORBinder       = &cborBinding{}
-)
+var HeaderBinderPool = sync.Pool{
+	New: func() any {
+		return &HeaderBinding{}
+	},
+}
+
+var RespHeaderBinderPool = sync.Pool{
+	New: func() any {
+		return &RespHeaderBinding{}
+	},
+}
+
+var CookieBinderPool = sync.Pool{
+	New: func() any {
+		return &CookieBinding{}
+	},
+}
+
+var QueryBinderPool = sync.Pool{
+	New: func() any {
+		return &QueryBinding{}
+	},
+}
+
+var FormBinderPool = sync.Pool{
+	New: func() any {
+		return &FormBinding{}
+	},
+}
+
+var URIBinderPool = sync.Pool{
+	New: func() any {
+		return &URIBinding{}
+	},
+}
+
+var XMLBinderPool = sync.Pool{
+	New: func() any {
+		return &XMLBinding{}
+	},
+}
+
+var JSONBinderPool = sync.Pool{
+	New: func() any {
+		return &JSONBinding{}
+	},
+}
+
+var CBORBinderPool = sync.Pool{
+	New: func() any {
+		return &CBORBinding{}
+	},
+}
+
+func GetFromThePool[T any](pool *sync.Pool) T {
```

Comment:
can we have an interface with the basic functions of a binder instead of any

`Name`, `Reset`, `Bind` if possible

Reply:
> can we have an interface with the basic functions of a binder instead of any
> 
> `Name`, `Reset`, `Bind` if possible

We could but some binders have different `Bind()` scheme such as URI vs Cookie etc.